### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by dlobanox**

### CODE REVIEW

File: build_scripts/build_runner.py [hunk 1]
```diff
@@ -157,40 +116,35 @@ def run(self):
                 self.cmd = ' && '.join(self.cmd)
 
             env = os.environ.copy()
+
+            if options:
+                self.cmd = self.cmd.format(**options)
+                if options['ENV']:
```
Comment (azhogov) [inline build_scripts/build_runner.py]: options.get('ENV') is safer

File: build_scripts/build_runner.py [hunk 2]
```diff
@@ -157,40 +116,35 @@ def run(self):
...
+                    env.update(options['ENV'])
```
Comment (azhogov) [inline build_scripts/build_runner.py]: should be after line 126, because it is dynamic date

File: build_scripts/build_runner.py [hunk 3]
```diff
@@ -397,18 +354,25 @@ def __init__(self, build_config_path, root_dir, build_type, product_type, build_
             "PACK_DIR": root_dir / "pack",
             "LOGS_DIR": root_dir / "logs",
             "BUILD_TYPE": build_type,  # sets from command line argument ('release' by default)
-            "CPU_CORES": multiprocessing.cpu_count()  # count of logical CPU cores
+            "CPU_CORES": multiprocessing.cpu_count(),  # count of logical CPU cores,
+            "VARS": {},
```
Comment (azhogov) [inline build_scripts/build_runner.py]: comment should be added

File: build_scripts/build_runner.py [hunk 4]
```diff
@@ -822,30 +825,46 @@ def main():
...
+        log.error('-' * 50)
+        log.error("%sING FAILED", args.stage.name)
+        exit(ErrorCode.CRITICAL.value)
 
 if __name__ == '__main__':
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    from common.helper import ErrorCode
+
     if platform.python_version_tuple() < ('3', '6'):
         print('\nERROR: Python 3.6 or higher required')
```
Comment (azhogov) [inline build_scripts/build_runner.py]: is required

File: build_scripts/build_runner.py [hunk 5]
```diff
@@ -822,30 +825,46 @@ def main():
...
+        build_state_file.write_text(json.dumps({'status': "FAIL"}))
+        log.error('-' * 50)
+        log.error("%sING FAILED", args.stage.name)
+        exit(ErrorCode.CRITICAL.value)
 
 if __name__ == '__main__':
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    from common.helper import ErrorCode
+
     if platform.python_version_tuple() < ('3', '6'):
```
Comment (azhogov) [inline build_scripts/build_runner.py]: probably should be shared too.

File: common/helper.py [hunk 6]
```diff
@@ -486,3 +509,29 @@ def update_json(check_type, success, output, json_path):
         with open(path, "w") as f:
             json.dump(new_data, f, indent=4, sort_keys=True)
     return True
+
+
+def cmd_exec(cmd, env=None, cwd=None, shell=True, log=None):
```
Comment (azhogov) [inline common/helper.py]: doc string should be added (regarding log especially)

File: build_scripts/build_runner.py [hunk 7]
```diff
@@ -557,59 +649,75 @@ def pack(self):
...
 
-        for action in self.actions[Stage.PACK]:
-            action.run()
+        if not self._run_build_config_actions(Stage.PACK):
+            return False
 
         if platform.system() == 'Windows':
             extension = "zip"
         elif platform.system() == 'Linux':
             extension = "tar"
```
Comment (azhogov) [inline build_scripts/build_runner.py]: please try zip on Linux too

Comment (dlobanox) [reply build_scripts/build_runner.py]: I will try it not while refactoring

**Pull request merged**